plugins/chat/assets/javascripts/discourse/components/chat-composer.js

```diff
@@ -147,11 +143,21 @@ export default class ChatComposer extends Component {
     );
   }
 
+  @action
+  didUpdateChannel() {
+    if (!this.args.channel) {
+      this.composer.message = null;
+    }
+
+    this.composer.message =
+      this.chatDraftsManager.get({ channelId: this.args.channel.id }) ||
+      ChatMessage.createDraftMessage(this.args.channel, {
+        user: this.currentUser,
+      });
+  }
+
```

Comment:
This is the main change of the PR as we aim to correctly update the composer message each time channel changes